platform-sdk/docs/proposals/001-tss-rosters-design-proposal/proposal.md

```diff
@@ -245,291 +518,180 @@ message RoundRosterPair {
-Process. It is essential to distinguish the Genesis Roster from the Candidate, Active, or Override rosters. The
-Genesis Roster is a special roster used for the sole purpose of bootstrapping a brand-new Genesis network; one in
-which there is no existing State, and the round number begins at 1. However, its structure is exactly the same as
-the Active or Candidate Rosters.
-
-The equivalent of the Genesis Roster in the current DevOps flow is the `config.txt` file. DevOps and Services may
-choose to create the Genesis Roster from the `config.txt` file or some other mechanism. The trigger for this
-workflow will be the presence of a Genesis Roster at start-up with no State on disk.
-
-A new method will be added to the `PlatformBuilder` that will be used by the App to set the Genesis Roster.
-In most cases, there will be no need to call the method on `PlatformBuilder` to set the Genesis Roster except in the
-events we need to bootstrap a Genesis Network.
+##### ReadableRosterStore
 
 ```java
-//in PlatformBuilder
-
-/*
-  Set the Genesis Roster for the network.
-  <p>
-  This method is used to set the optional Genesis Roster for the network.
-  The Genesis Roster is a special roster used for the sole-purpose of bootstrapping a network.
-  The Genesis Roster, if set, is immutable and will be used to start a Genesis network Process or Network Transplant process.
-  <p>
-  @param genesisRoster the Genesis Roster for the network.
+/**
+ * Read-only implementation for accessing rosters states.
  */
-PlatformBuilder withGenesisRoster(@NonNull final Roster genesisRoster);
-```
-
-The logic to determine whether to invoke this method could be determined by some mechanism Services deems appropriate.
-
-#### Override Roster
+public interface ReadableRosterStore {
+    /**
+     * Gets the candidate roster if found in state or null otherwise.
+     * Note that state commits are buffered,
+     * so it is possible that a recently stored candidate roster is still in the batched changes and not yet committed.
+     * Therefore, callers of this API must bear in mind that an immediate call after storing a candidate roster may return null.
+     *
+     * @return the candidate roster
+     */
+    @Nullable
+    Roster getCandidateRoster();
 
-An `Override Roster` is an optional Roster that DevOps may provide for the explicit purpose of starting a network using
-a specific state, with a specific set of nodes, such as during a Network Transplant Process. A common example of
-this is Mainnet State being transplanted into a testing network.
+    /**
+     * Gets the active roster.
+     * Returns the active roster iff:
+     *      the roster state singleton is not null
+     *      the list of round roster pairs is not empty
+     *      the first round roster pair exists
+     *      the active roster hash is present in the roster map
+     * otherwise returns null.
```

Comment:
If the roster is valid, most of these conditions appear to be met. Is it necessary to list them here? (same below)